x-pack/test/api_integration/apis/management/snapshot_restore/snapshot_restore.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { registerEsHelpers } from './lib';
+
+const API_BASE_PATH = '/api/snapshot_restore';
+const REPO_NAME = 'test_repo';
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+
+  const { createRepository, createPolicy, deletePolicy, cleanupPolicies } = registerEsHelpers(
+    getService
+  );
+
+  describe('Snapshot Lifecycle Management', function () {
+    before(async () => {
+      try {
+        await createRepository(REPO_NAME);
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.log('[Setup error] Error creating repository');
+        throw err;
+      }
+    });
+
+    after(async () => {
+      await cleanupPolicies();
+    });
+
+    describe('Create', () => {
+      const POLICY_NAME = 'test_create_policy';
+      const REQUIRED_FIELDS_POLICY_NAME = 'test_create_required_fields_policy';
+
+      after(async () => {
+        // Clean up any policies created in test cases
+        await Promise.all([POLICY_NAME, REQUIRED_FIELDS_POLICY_NAME].map(deletePolicy)).catch(
+          (err) => {
+            // eslint-disable-next-line no-console
+            console.log(`[Cleanup error] Error deleting policies: ${err.message}`);
+            throw err;
+          }
+        );
+      });
+
+      it('should create a SLM policy', async () => {
+        const { body } = await supertest
+          .post(`${API_BASE_PATH}/policies`)
+          .set('kbn-xsrf', 'xxx')
+          .send({
+            name: POLICY_NAME,
+            snapshotName: 'my_snapshot',
+            schedule: '0 30 1 * * ?',
+            repository: REPO_NAME,
+            config: {
+              indices: ['my_index'],
+              ignoreUnavailable: true,
+              partial: false,
+              metadata: {
+                meta: 'my_meta',
+              },
+            },
+            retention: {
+              expireAfterValue: 1,
+              expireAfterUnit: 'd',
+              maxCount: 10,
+              minCount: 5,
+            },
+            isManagedPolicy: false,
+          })
+          .expect(200);
+
+        expect(body).to.eql({
```

Comment:
For the "create" and "update" tests, would it be worth fetching the policy and verifying that it's been created/updated? Otherwise we know that our API has responded as if everything's great, but we haven't verified that integration with ES behaved as expected.

I also don't see a "delete" test, should we add one?

Reply:
> For the "create" and "update" tests, would it be worth fetching the policy and verifying that it's been created/updated? Otherwise we know that our API has responded as if everything's great, but we haven't verified that integration with ES behaved as expected.

Good point! Added this check for the tests.

> I also don't see a "delete" test, should we add one?

I'm going to hold off on this for now, as it feels outside the scope of the initial PR. There are [some tests](https://github.com/elastic/kibana/blob/master/x-pack/plugins/snapshot_restore/server/routes/api/policy.test.ts#L180) already that run using mocks.